### ISSUE TITLE
Crash with "list index out of range" when scrolling a recycleview with scrollbar

### ISSUE THREAD

**Issue by snuq** [kivy/kivy#7256]
**Software Versions**
* Python: 3.8.6
* OS: Windows 10
* Kivy: 2.0.0
* Kivy installation method: pip

**Describe the bug**
This may only happen on windows, a user running debian reported that it was not happening. Able to confirm that it happens on on python 3.7.8 as well.

When using a recycleview as in this example code:
```

from kivy.app import App
from kivy.lang.builder import Builder
KV = """
RecycleView:
    scroll_type: ['bars', 'content']
    bar_width: 20
    data: [{"text": "Label "+str(x)} for x in range(20)]
    viewclass: 'Button'
    RecycleBoxLayout:
        spacing: 10
        default_size: None, dp(80)
        default_size_hint: 1, None
        orientation: 'vertical'
        size_hint_y: None
        height: self.minimum_height
"""
class Test(App):
    def build(self):
        return Builder.load_string(KV)
Test().run()
```
kivy will crash if you try and scroll with the scrollbar, tho it will not crash if the content is scrolled, THEN the scrollbar is moved.

The traceback from the error: 
```
Traceback (most recent call last):
   File "main.py", line 20, in <module>
     Test().run()
   File "C:\Python38\lib\site-packages\kivy\app.py", line 950, in run
     runTouchApp()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 582, in runTouchApp
     EventLoop.mainloop()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 347, in mainloop
     self.idle()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 391, in idle
     self.dispatch_input()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 342, in dispatch_input
     post_dispatch_input(*pop(0))
   File "C:\Python38\lib\site-packages\kivy\base.py", line 301, in post_dispatch_input
     wid.dispatch('on_touch_move', me)
   File "kivy\_event.pyx", line 709, in kivy._event.EventDispatcher.dispatch
   File "C:\Python38\lib\site-packages\kivy\uix\scrollview.py", line 855, in on_touch_move
     if self.dispatch('on_scroll_move', touch):
   File "kivy\_event.pyx", line 709, in kivy._event.EventDispatcher.dispatch
   File "C:\Python38\lib\site-packages\kivy\uix\scrollview.py", line 908, in on_scroll_move
     self.effect_x.update(touch.x)
   File "C:\Python38\lib\site-packages\kivy\effects\scroll.py", line 116, in update
     self.displacement += abs(val - self.history[-1][1])
 IndexError: list index out of range
```

edit: somehow github was hiding the first line of the code

**Comment by tshirtman** [comment]
hm, looks like the issue that should have been fixed by https://github.com/kivy/kivy/pull/7213

**Comment by matham** [comment]
Are you sure you tried with 2.0.0? Because I just tried it and could not replicate the issue on Windows. And this should really not be OS specific.

I ran the code and then tried scrolling with the scrollbar.

**Comment by snuq** [comment]
its definitely kivy 2.0.0, kivy 1.1.1 doesnt crash in the same way. 

full console log:
```
C:\Users\snu\Desktop>C:\Python38\python.exe main.py
[INFO   ] [Logger      ] Record log in C:\Users\snu\.kivy\logs\kivy_20-12-10_3.txt
[INFO   ] [deps        ] Successfully imported "kivy_deps.angle" 0.3.0
[INFO   ] [deps        ] Successfully imported "kivy_deps.glew" 0.3.0
[INFO   ] [deps        ] Successfully imported "kivy_deps.sdl2" 0.3.1
[INFO   ] [Kivy        ] v2.0.0
[INFO   ] [Kivy        ] Installed at "C:\Python38\lib\site-packages\kivy\__init__.py"
[INFO   ] [Python      ] v3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]
[INFO   ] [Python      ] Interpreter at "C:\Python38\python.exe"
[INFO   ] [Factory     ] 186 symbols loaded
[INFO   ] [ImageLoaderFFPy] Using ffpyplayer 4.3.2
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil, img_ffpyplayer
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] GLEW initialization succeeded
[INFO   ] [GL          ] Backend used <glew>
[INFO   ] [GL          ] OpenGL version <b'4.6.0 NVIDIA 452.11'>
[INFO   ] [GL          ] OpenGL vendor <b'NVIDIA Corporation'>
[INFO   ] [GL          ] OpenGL renderer <b'GeForce RTX 2060/PCIe/SSE2'>
[INFO   ] [GL          ] OpenGL parsed version: 4, 6
[INFO   ] [GL          ] Shading version <b'4.60 NVIDIA'>
[INFO   ] [GL          ] Texture max size <32768>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [Text        ] Provider: sdl2
[INFO   ] [Base        ] Start application main loop
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [Base        ] Leaving application in progress...
 Traceback (most recent call last):
   File "main.py", line 20, in <module>
     Test().run()
   File "C:\Python38\lib\site-packages\kivy\app.py", line 950, in run
     runTouchApp()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 582, in runTouchApp
     EventLoop.mainloop()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 347, in mainloop
     self.idle()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 391, in idle
     self.dispatch_input()
   File "C:\Python38\lib\site-packages\kivy\base.py", line 342, in dispatch_input
     post_dispatch_input(*pop(0))
   File "C:\Python38\lib\site-packages\kivy\base.py", line 301, in post_dispatch_input
     wid.dispatch('on_touch_move', me)
   File "kivy\_event.pyx", line 709, in kivy._event.EventDispatcher.dispatch
   File "C:\Python38\lib\site-packages\kivy\uix\scrollview.py", line 855, in on_touch_move
     if self.dispatch('on_scroll_move', touch):
   File "kivy\_event.pyx", line 709, in kivy._event.EventDispatcher.dispatch
   File "C:\Python38\lib\site-packages\kivy\uix\scrollview.py", line 908, in on_scroll_move
     self.effect_x.update(touch.x)
   File "C:\Python38\lib\site-packages\kivy\effects\scroll.py", line 116, in update
     self.displacement += abs(val - self.history[-1][1])
 IndexError: list index out of range
```

fresh install of python also

**Comment by snuq** [comment]
![test](https://user-images.githubusercontent.com/13342757/101858096-bd3e5000-3b1d-11eb-8ff0-9ba70411cc4a.gif)
demonstration of the crash

**Comment by matham** [comment]
Oh yeah, my bad. I thought it was merged in 2.0.0, but spaced on the fact I merged it just after 2.0.0 so it is fixed in master, which is what I was testing locally.

The PR was submitted too close to 2.0.0 to have it merged in 2.0.0.
